nncf/experimental/torch/fx/engine.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2024 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Tuple, Union
+
+import torch
+from torch import nn
+
+from nncf.common.engine import Engine
+
+
+class FXEngine(Engine):
```

Comment:
It is possible, but `PTEngine` is calling `.eval()` method on the model which is not available in `torch.fx.GraphModule` after the quantization. We either could add condition to `PTEngine` to prevent `.eval()` calling or keep a separate implementation. What would you prefere?

Reply:
Done, please check